### DISCUSSION TITLE
“Soft off” vs “Hard off” HVAC?

### DISCUSSION THREAD

**Discussion by colmbuckley** [alandtse/tesla#92] [Q&A]
I've been noticing something strange when writing automations using this integration:
- When I turn off my car's HVAC using the Tesla app, it turns off as expected, but when I then get into the car to drive it activates again automatically.
- When I turn it off using `climate.set_hvac_mode` with `hvac_mode: 'off'`, again it turns off as expected, but when I get into the car later, it does **not** turn back on automatically.

I'm thinking there's some subtle difference here between “soft off” (which turns back on automatically) and “hard off” (which stays off until manually turned on). I'd like to be able to specify these in HASS; anyone found a way to make this happen? Various fiddlings with the preset modes and climate functions haven't revealed much.

(Car is a 2016 Model S, if that makes a difference.)

**Comment by alandtse** [comment]
It'd be interesting to have someone sniff the official app to see if it's using another endpoint or parameter. Or perhaps it's just sneakily reenabling the climate later like at midnight and the car is just turning it back off after a set interval due to inactivity.
